### ISSUE TITLE
Delete related torrents when deleting a folder or video file

### ISSUE THREAD

**Issue by dfernandezm** [dfernandezm/download-utils#84]
This should be done using a transactional block, so both things are deleted at the same time. The typical cases would be:
- Deleting a whole TV Show folder - delete all the torrents related with it (user renamedPath)
- Deleting a Season - delete all the torrents related with it
- Deleting an episode - delete the torrent related with it

It would help creating a new entity MediaAsset which will hold each torrent asset downloaded (files)

**Comment by fniewijk** [comment]
What about deleting the files dependent on the torrent you are deleting?

**Comment by dfernandezm** [comment]
Yes, that would be how it should work, but in this case we have a screen with all our folders in a list, we want to delete one, so it has to delete the filesystem hierarchy and the torrent entry.

The UI here clearly should be the opposite, have a list of TV Shows / Movies, click delete and delete both torrent and file hierarchy.

One big TODO to support this is actually storing TV Show and Movies data en the DB (i.e having this model in place). All the needed info can be retrieved automatically using a third party API like trakt.tv, users don't need to fill anything.
